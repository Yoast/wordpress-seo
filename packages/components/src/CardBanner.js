import React, { Fragment } from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

/* Yoast dependencies */
import { colors } from "@yoast/style-guide";

const BannerContents = styled.span`
	position: absolute;
	
	top: 8px;
	left: -8px;
	
	font-weight: 500;
	color: ${ props => props.textColor };
	line-height: 16px;
	
	background-color: ${ props => props.backgroundColor };
	padding: 8px 16px;
	box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.2);
`;

const BannerTriangle = styled.span`
	position: absolute;
	
	top: 40px;
	left: -8px;
	
	/* This code makes the triangle. */
	border-top: 8px solid ${ colors.$color_purple_dark };
	border-left: 8px solid transparent;
`;

/**
 * The banner component.
 * @param {Object} props The props.
 * @returns {ReactElement} The banner element.
 * @constructor
 */
export default function Banner( props ) {
	return <Fragment>
		<BannerContents backgroundColor={ props.backgroundColor } textColor={ props.textColor }>
			{ props.children }
		</BannerContents>
		<BannerTriangle />
	</Fragment>;
}

Banner.propTypes = {
	backgroundColor: PropTypes.string,
	textColor: PropTypes.string,
	children: PropTypes.any,
};

Banner.defaultProps = {
	backgroundColor: colors.$color_pink_dark,
	textColor: colors.$color_white,
	children: null,
};
