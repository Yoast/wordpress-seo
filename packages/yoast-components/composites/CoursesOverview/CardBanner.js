export { CardBanner as default } from "@yoast/components";
