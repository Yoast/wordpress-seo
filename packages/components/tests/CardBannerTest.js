import React from "react";
import renderer from "react-test-renderer";
import CardBanner from "../src/CardBanner";

test( "The empty CardBanner matches the snapshot", () => {
	const component = renderer.create(
		<CardBanner />
	);

	const tree = component.toJSON();
	expect( tree ).toMatchSnapshot();
} );

test( "The CardBanner with specific props matches the snapshot", () => {
	const banner = {
		backgroundColor: "purple",
		textColor: "yellow",
		text: "Free Trial Available.",
	};
	const component = renderer.create(
		<CardBanner { ...banner }>
			{ banner.text }
		</CardBanner>
	);

	const tree = component.toJSON();
	expect( tree ).toMatchSnapshot();
} );

