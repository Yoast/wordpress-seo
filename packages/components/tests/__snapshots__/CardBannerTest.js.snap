// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The CardBanner with specific props matches the snapshot 1`] = `
Array [
  .c0 {
  position: absolute;
  top: 8px;
  left: -8px;
  font-weight: 500;
  color: yellow;
  line-height: 16px;
  background-color: purple;
  padding: 8px 16px;
  box-shadow: 0 2px 4px 0 rgba(0,0,0,0.2);
}

<span
    className="c0"
  >
    Free Trial Available.
  </span>,
  .c0 {
  position: absolute;
  top: 40px;
  left: -8px;
  border-top: 8px solid #6c2548;
  border-left: 8px solid transparent;
}

<span
    className="c0"
  />,
]
`;

exports[`The empty CardBanner matches the snapshot 1`] = `
Array [
  .c0 {
  position: absolute;
  top: 8px;
  left: -8px;
  font-weight: 500;
  color: #fff;
  line-height: 16px;
  background-color: #a4286a;
  padding: 8px 16px;
  box-shadow: 0 2px 4px 0 rgba(0,0,0,0.2);
}

<span
    className="c0"
  />,
  .c0 {
  position: absolute;
  top: 40px;
  left: -8px;
  border-top: 8px solid #6c2548;
  border-left: 8px solid transparent;
}

<span
    className="c0"
  />,
]
`;
